addon/components/responsive-toolbar.ts

```diff
@@ -0,0 +1,148 @@
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+import { modifier } from 'ember-modifier';
+import { tracked } from 'tracked-built-ins';
+
+export default class ResponsiveToolbar extends Component {
+  toolbar?: HTMLElement;
+
+  main: {
+    toolbar?: HTMLElement;
+    dropdown?: HTMLElement;
+    enableDropdown: boolean;
+    showDropdown: boolean;
+  } = tracked({
+    enableDropdown: false,
+    showDropdown: false,
+  });
+
+  side: {
+    toolbar?: HTMLElement;
+    dropdown?: HTMLElement;
+    enableDropdown: boolean;
+    showDropdown: boolean;
+  } = tracked({
+    enableDropdown: false,
+    showDropdown: false,
+  });
+
+  setUpToolbar = modifier(
+    (element: HTMLElement) => {
+      const observer = new ResizeObserver(this.handleResize.bind(this));
+      observer.observe(element);
+      this.toolbar = element;
+    },
+    { eager: false }
+  );
+
+  setUpMainToolbar = modifier(
+    (element: HTMLElement) => {
+      this.main.toolbar = element;
+      // Call handleResize to ensure the toolbar is correctly initialized
+      this.handleResize();
+    },
+    { eager: false }
+  );
+
+  setUpSideToolbar = modifier(
+    (element: HTMLElement) => {
+      this.side.toolbar = element;
+      // Call handleResize to ensure the toolbar is correctly initialized
+      this.handleResize();
+    },
+    { eager: false }
+  );
+
+  setUpMainDropdown = modifier(
+    (element: HTMLElement) => {
+      this.main.dropdown = element;
+      // Call handleResize to ensure the toolbar is correctly initialized
+      this.handleResize();
+    },
+    { eager: false }
+  );
+
+  setUpSideDropdown = modifier(
+    (element: HTMLElement) => {
+      this.side.dropdown = element;
+      // Call handleResize to ensure the toolbar is correctly initialized
+      this.handleResize();
+    },
+    { eager: false }
+  );
+
+  @action
+  toggleMainDropdown() {
+    this.main.showDropdown = !this.main.showDropdown;
+    if (this.main.showDropdown) {
+      this.side.showDropdown = false;
+    }
+  }
+
+  @action
+  toggleSideDropdown() {
+    this.side.showDropdown = !this.side.showDropdown;
+    if (this.side.showDropdown) {
+      this.main.showDropdown = false;
+    }
+  }
+
+  isOverflowing = () => {
```

Comment:
any reason this is not a getter? (if there is, would like a code comment explaining why)

Reply:
Not really :stuck_out_tongue: